### PR TITLE
Fix some race conditions with loading plugin config and migrate to the cc rules

### PR DESCRIPTION
**Pull request by Tatskaari**
This fixes some race conditions around architecture subrepos that were being hit because of the preloaded subinclude.

### CODE REVIEW

File: src/core/state.go [hunk 1]
```diff
@@ -216,6 +216,52 @@ type BuildState struct {
...
+				return
+			}
+			if err = validateSubrepoNameAndPluginConfig(state.Config, state.RepoConfig, subrepo); err != nil {
+				return
+			}
+		}
+
+		go state.Parser.Init(state)
+	})
+	return
```
Comment (samwestmoreland) [inline src/core/state.go:259]: Do we need to return nil or error here depending on how `Do` went?

File: src/core/subrepo.go [hunk 2]
```diff
@@ -25,20 +24,30 @@ type Subrepo struct {
 	Arch cli.Arch
 	// True if this subrepo was created for a different architecture
 	IsCrossCompile bool
-	// loadConfig is used to control when we load plugin configuration. We need access to the subrepo itself to do this
-	// so it happens at build time.
-	loadConfig            sync.Once
+	//
```
Comment (samwestmoreland) [inline src/core/subrepo.go:27]: Missing comment here I think!

**Review by samwestmoreland — APPROVED** [review]
Nice! Think this is a lot neater. Just a couple of minor comments.

File: src/core/subrepo.go (hunk 2 above)
Comment (Tatskaari) [reply src/core/subrepo.go:27]: No comment

File: src/core/state.go (hunk 1 above)
Comment (Tatskaari) [reply src/core/state.go:259]: Ah, this is a go thing. We name the return variable, so when we can set it in the closure. When we return like this, we return that variable from the function. It's a bit weird but useful in this sort of situation.

Comment (samwestmoreland) [reply src/core/state.go:259]: Oh I see, ok that makes sense then. Thanks!

**Pull request merged**